03-interpolated-defaults.tf

```diff
@@ -1,10 +1,10 @@
 locals {
 
   sub_env        = var.environment == "ptlsbox" ? "ptl-sbox" : var.environment
-  aad_group_name = contains(["ptlsbox", "ptl"], var.environment) ? "DTS Contributors (sub:dts-sharedservices${local.sub_env})" : "DTS Contributors (sub:dts-sharedservices-${local.sub_env})"
-
   
-  project = length(regexall(".*sds.*", lower(var.developers_group))) > 0 ? "sds" : "cft"
+  aad_group_name = var.environment == "ptl" && local.project == "cft" ? "DTS Contributors (sub:dts-cftptl-intsvc)" : (var.environment == "ptlsbox" && local.project == "cft" ? "DTS Contributors (sub:dts-cftsbox-intsvc)" : var.environment == "ptl" && local.project == "sds" ? "DTS Contributors (sub:dts-sharedservicesptl)" : (var.environment == "ptlsbox" && local.project == "sds" ? "DTS Contributors (sub:dts-sharedservicesptl-sbox)" : (!contains(["ptl","ptlsbox"], var.environment) && local.project == "sds" ? "DTS Contributors (sub:dts-sharedservices-${local.sub_env})" : "DTS Contributors (sub:dcd-cftapps-${local.sub_env})")))
```

Comment:
how was this working before?

I can't see any CFT specific logic.

Anyway isn't this just the active subscriptions displayname lowercased?
You could get the exact algorithm from the azure-enterprise repository

Reply:
The project local value was being derived from the variable developers group which, in aks-cft-deploy, is set to  `DTS Operations env: $env` which has no mention of `cft` or `sds` so this local always resolves to `sds`. This local is used to determine the keyvault name [here](https://github.com/hmcts/aks-module-genesis/blob/7af7c076fbe1e3db172b1b819360ca88ad6d090a/20-key-vault.tf#L7).

The aad_group_name local currently has sharedservices groups hardcoded so, at the minute, cft keyvaults have the sharedservices contributors groups granted access but not the equivalent groups for cft. Not sure why it was left this way.

Reply:
I would start by changing the project variable to be `businessArea` and then it should be passed down from the consuming repositories.

I think you should be able to determine the group name sensibly then

Reply:
Tried to simplify it somewhat using businessArea tag. Getting tag value from https://github.com/hmcts/aks-cft-deploy/pull/457/files#diff-0aa6a4ae14301983310f5a332fd11d5c51ed8c5848613358af32d90599a80331R6